.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,35 @@
+name: golangci-lint
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened]
+
+# Remove all permissions from GITHUB_TOKEN except metadata.
+permissions: {}
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        working-directory:
+          - ""
+          - test
+          - apis
+          - pkg/hardwareutils
```

Comment:
Should we have the `if:` conditional here too like in other workflows?

Reply:
I think not. This workflow should work well also in forks, right?

Reply:
Good question, haven't thought about it, but I am leaning towards @lentzi90's comment that it should work with forks also. So we can keep it as it is now.

Reply:
It doesn't have anything metal3 specific like uploads etc, so it should be fine.